### PR TITLE
[mergify] Update rules for matrix build, fix non-semver packages

### PR DESCRIPTION
**Pull request by Nuru**
## what
- [mergify] Update rules to check for success of for new matrix builds
- Convert non-conforming (to semver) package versions to semver
  - gosu
  - lazydocker
  - lectl
- Disable auto-update of `cloudposse-atlantis` because it does not follow semver. 

## why
- "Checks" for PRs changed significantly with #689, old rules were no longer correct
- We now use semver to determine auto-updates, so we need our package versions to be semver compliant

### CODE REVIEW

File: .github/mergify.yml
Comment (osterman) [inline .github/mergify.yml:23]: I think we said it should approve only when the files are in `vendor/*`

With this logic, it will approve even PRs by a `CODEOWNER` to `tasks/`, which should still have an approval.

Comment (osterman) [inline .github/mergify.yml:23]:
```suggestion
      - "#files<=14"
      - "files~=^vendor/[^/]+/VERSION"
```

Comment (Nuru) [reply .github/mergify.yml:23]: This rule just means that one of the changed files must be a `VERSION`. Not what we want.

**Review by osterman — APPROVED** [review]

**Pull request merged**
